File: basket-service/src/main/java/com/testShop/microservices/basketservice/configuration/RabbitConfiguration.java
package com.testShop.microservices.basketservice.configuration;

import org.springframework.amqp.core.AmqpAdmin;
import org.springframework.amqp.core.Queue;
import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
import org.springframework.amqp.rabbit.connection.ConnectionFactory;
import org.springframework.amqp.rabbit.core.RabbitAdmin;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RabbitConfiguration {

//    @Value("${queue.name}")
//    private String queueName;
//
//    private final String LOCALHOST = "localhost";
//
//    @Bean
//    public ConnectionFactory connectionFactory() {
//        CachingConnectionFactory connectionFactory =
//            new CachingConnectionFactory(LOCALHOST);
//        return connectionFactory;
//    }
//
//    @Bean
//    public AmqpAdmin amqpAdmin() {
//        return new RabbitAdmin(connectionFactory());
//    }
//
//    @Bean
//    public RabbitTemplate rabbitTemplate() {
//        return new RabbitTemplate(connectionFactory());
//    }
//
//    @Bean
//    public Queue myQueue1() {
//        return new Queue(queueName);
//    }
}

File: kotlin-listener-service/src/main/kotlin/com/testShop/microservices/kotlinlistenerservice/configuration/RabbitConfiguration.kt
package com.testShop.microservices.kotlinlistenerservice.configuration

import org.springframework.amqp.core.AmqpAdmin
import org.springframework.amqp.core.Queue
import org.springframework.amqp.rabbit.connection.CachingConnectionFactory
import org.springframework.amqp.rabbit.connection.ConnectionFactory
import org.springframework.amqp.rabbit.core.RabbitAdmin
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class RabbitConfiguration {
    @Value("\${queue.name}")
    private val queueName: String? = null
    private val LOCALHOST = "localhost"

    @Bean
    open fun connectionFactory(): ConnectionFactory {
        return CachingConnectionFactory(LOCALHOST)
    }

    @Bean
    open fun amqpAdmin(): AmqpAdmin {
        return RabbitAdmin(connectionFactory())
    }

    @Bean
    open fun rabbitTemplate(): RabbitTemplate {
        return RabbitTemplate(connectionFactory())
    }

    @Bean
    open fun myQueue1(): Queue {
        return Queue(queueName)
    }
}
